lambdas/src/main/kotlin/scorcerer/server/resources/MatchResource.kt

```diff
@@ -21,10 +22,31 @@ class MatchResource : MatchApi() {
     }
 
     override fun listMatches(requesterUserId: String, filterType: String?): List<Match> {
-        return listOf(
-            Match("England", "France", "12345"),
-            Match("Scotland", "Germany", "12346"),
-        )
+        val matches: List<Match>
+        if (filterType != null) {
+            matches = transaction {
+                MatchTable.selectAll().where { MatchTable.state eq MatchState.valueOf(filterType.uppercase()) }
+                    .map { row ->
+                        Match(
+                            row[MatchTable.homeTeamId].toString(),
+                            row[MatchTable.awayTeamId].toString(),
+                            row[MatchTable.id].toString(),
+                        )
+                    }
+            }
+        } else {
+            matches = transaction {
+                MatchTable.selectAll()
+                    .map { row ->
+                        Match(
+                            row[MatchTable.homeTeamId].toString(),
+                            row[MatchTable.awayTeamId].toString(),
+                            row[MatchTable.id].toString(),
+                        )
+                    }
+            }
+        }
+        return matches
     }
```

Comment:
You're flying, thanks for adding these!

We can probably refactor this a bit. First off you can just return the if statement. And then it probably makes sense to pull out the map so that it's not duplicated. Something like this

```
override fun listMatches(requesterUserId: String, filterType: String?): List<Match> =
        if (filterType.isNullOrBlank()) {
            transaction { MatchTable.selectAll() }
        } else {
            transaction { MatchTable.selectAll().where { MatchTable.state eq MatchState.valueOf(filterType.uppercase()) } }
        }.map { row ->
            Match(
                row[MatchTable.homeTeamId].toString(),
                row[MatchTable.awayTeamId].toString(),
                row[MatchTable.id].toString())
        }
```

Reply:
Ahh nice will do that thought you could probably do something like that

Reply:
Ahh the map needs to be inside the transaction. But could probs do the if else inside the transaction